### PR TITLE
[FIX] Allow preprocessed data in native space in `load_confounds`

### LINKED ISSUE #3479: Less restrictive of `load_confounds` file mathcing pattern
The `load_confounds` function does a lazy search based on the image file name to find other required files from a fMRIPrep derivative. Currently it is restricted to outputs that are mapped to a group template. This raise an issue when people wants to use files in subject's native space. In this case the `space-` entity would not be present. 

### Benefits to the change
Load data preprocessed in native space with ease.

### Pseudocode for the new behavior, if applicable
It will not raise errors at this specific line here:
https://github.com/nilearn/nilearn/blob/9c29e3376243b643a72f78b8b98a56d12c2e99bb/nilearn/interfaces/fmriprep/load_confounds_utils.py#L92
No changes needed for the user facing side.

### PR DESCRIPTION
**Pull request by htwangtw**
Closes #3479.

Changes proposed in this pull request:

- Use `nilearn.interfaces.bids.parse_bids_filename` to parse and construct file names that fits fMRIPrep doc better
- Adapt BIDS compatible test file name

### CODE REVIEW

File: nilearn/interfaces/fmriprep/load_confounds_utils.py [hunk 1]
```diff
@@ -89,15 +89,22 @@ def _get_file_name(nii_file):
     """Construct the raw confound file name from processed functional data."""
     if isinstance(nii_file, list):  # catch gifti
         nii_file = nii_file[0]
-    suffix = "_space-" + nii_file.split("space-")[1]
+    entities = parse_bids_filename(nii_file)
+    subject_label = f"sub-{entities['sub']}"
+    if "session" in entities:
```
Comment (Remi-Gau) [inline nilearn/interfaces/fmriprep/load_confounds_utils.py:94]: Still wrapping my head around how nilearn deals with BIDS things but isn't that supposed to be 

```python
if "ses" in entities:
```

Comment (htwangtw) [reply nilearn/interfaces/fmriprep/load_confounds_utils.py:94]: Yes, and the test is missing this line. Will fix it

File: nilearn/interfaces/fmriprep/load_confounds_utils.py [hunk 2]
```diff
@@ -10,13 +10,14 @@
...
 )
+from nilearn.interfaces.bids import parse_bids_filename
 
 
 img_file_patterns = {
     "aroma": "_desc-smoothAROMAnonaggr_bold",
-    "nii.gz": "_space-.*_desc-preproc_bold.nii.gz",
-    "dtseries.nii": "_space-.*_bold.dtseries.nii",
-    "func.gii": "_space-.*_hemi-[LR]_bold.func.gii",
+    "nii.gz": "(_space-.*)?_desc-preproc_bold\.nii\.gz",
```
Comment (bthirion) [inline nilearn/interfaces/fmriprep/load_confounds_utils.py:18]: Just curious: what is the effects of `\` in the string ?

**Review by bthirion — APPROVED** [review]
LGTM, thx.

File: nilearn/interfaces/fmriprep/load_confounds_utils.py (hunk 2 above)
Comment (ymzayek) [reply nilearn/interfaces/fmriprep/load_confounds_utils.py:18]: I think it escapes the . as a wildcard character in regular expressions

File: nilearn/interfaces/fmriprep/load_confounds_utils.py [hunk 3]
```diff
@@ -89,15 +90,22 @@ def _get_file_name(nii_file):
...
     # fmriprep has changed the file suffix between v20.1.1 and v20.2.0 with
     # respect to BEP 012.
     # cf. https://neurostars.org/t/naming-change-confounds-regressors-to-confounds-timeseries/17637 # noqa
     # Check file with new naming scheme exists or replace,
     # for backward compatibility.
     confounds_raw_candidates = [
-        nii_file.replace(suffix, "_desc-confounds_timeseries.tsv"),
-        nii_file.replace(suffix, "_desc-confounds_regressors.tsv"),
+        nii_file.replace(img_filename, f"{subject_label}_{specifiers}_desc-confounds_timeseries.tsv"),
+        nii_file.replace(img_filename, f"{subject_label}_{specifiers}_desc-confounds_regressors.tsv"),
```
Comment (ymzayek) [inline nilearn/interfaces/fmriprep/load_confounds_utils.py:108]:
```suggestion
~         nii_file.replace(
~             img_filename,
+             f"{subject_label}_{specifiers}_desc-confounds_timeseries.tsv"
+         ),
+         nii_file.replace(
+             img_filename,
+             f"{subject_label}_{specifiers}_desc-confounds_regressors.tsv"
+         ),
```

PEP8 line length

**Review by ymzayek — COMMENTED** [review]
LGTM @htwangtw! I think this is worth a whatsnew entry

**Comment by ymzayek** [comment]
@htwangtw since I made a prerelease earlier this week the `latest.rst` file changed. Can you rebase on main and move the whatsnew entry to the new section?

**Pull request merged**